libs/feature/map/src/lib/geocoding.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable, Inject, InjectionToken } from '@angular/core'
+import {
+  queryGeoadmin,
+  GeoadminOptions,
+  GeocodingResult,
+  queryGeonames,
+  GeonamesOptions,
+  DataGouvFrOptions,
+  queryDataGouvFr,
+} from '@geospatial-sdk/geocoding'
+import { from, Observable, throwError } from 'rxjs'
+import { catchError } from 'rxjs/operators'
+
+export const GEOCODING_PROVIDER = new InjectionToken('geocoding-provider')
+
+export interface GeocodingProvider {
+  id: 'geoadmin' | 'geonames' | 'data-gouv-fr'
+  options: GeoadminOptions | GeonamesOptions | DataGouvFrOptions
+}
```

Comment:
I wonder if having something more like `['geoadmin', { ...options }]` would be easier to handle.

So the type would be:
```typescript
type GeoadminGeocodingProvider = [
  'geoadmin',
  GeoadminOptions
]
type GeonamesGeocodingProvider = [
  'geonames',
  GeonamesOptions
]
type DataGouvFrGeocodingProvider = [
  'data-gouv-fr',
  DataGouvFrOptions
]
export type GeocodingProvider = GeoadminGeocodingProvider | GeonamesGeocodingProvider | DataGouvFrGeocodingProvider
```

What do you think?

Reply:
Sounds good, will implement this suggestion.